vayesta/ewf/urdm.py

```diff
@@ -208,11 +211,494 @@ def make_rdm1_ccsd_global_wf(emb, ao_basis=False, with_mf=True, t_as_lambda=Fals
+                    cx2_occ_a = np.dot(cs_occ_a, cx2_occ_a)
+                    cx2_vir_a = np.dot(cs_vir_a, cx2_vir_a)
+                    cx2_occ_b = np.dot(cs_occ_b, cx2_occ_b)
+                    cx2_vir_b = np.dot(cs_vir_b, cx2_vir_b)
+                    if not late_t2_sym:
+                        dvoa += einsum('ai,Ii,Aa->AI', dvoxa, cx2_occ_a, cx2_vir_a)
+                        dvob += einsum('ai,Ii,Aa->AI', dvoxb, cx2_occ_b, cx2_vir_b)
+                    else:
+                        dvoa += dot(cx2_vir_a, dvoxa1, cx2_frag_a.T, cs_occ_a.T)
+                        dvoa += dot(cx2_vir_a, dvoxa2, cx2_occ_a.T)
+                        dvob += dot(cx2_vir_b, dvoxb1, cx2_frag_b.T, cs_occ_b.T)
+                        dvob += dot(cx2_vir_b, dvoxb2, cx2_occ_b.T)
+
+    if mpi:
+        rma.clear()
+        dooa, doob, dvoa, dvob, dvva, dvvb = mpi.nreduce(dooa, doob, dvoa, dvob, dvva, dvvb, target=mpi_target, logfunc=emb.log.timingv)
+        # Make sure no more MPI calls are made after returning some ranks early!
+        if mpi_target not in (None, mpi.rank):
+            return None
+
+    if with_t1:
+        xt1a = -dooa.T.copy()
+        xt1b = -doob.T.copy()
+        xt2a = dvva.copy()
+        xt2b = dvvb.copy()
+
+        dooa -= einsum('ie,je->ij', l1a, t1a)
+        doob -= einsum('ie,je->ij', l1b, t1b)
+
+        dvva += einsum('ma,mb->ab', t1a, l1a)
+        dvvb += einsum('ma,mb->ab', t1b, l1b)
+
+        xt2a += einsum('ma,me->ae', t1a, l1a)
+        dvoa -= einsum('mi,ma->ai', xt1a, t1a)
+        dvoa -= einsum('ie,ae->ai', t1a, xt2a)
+        dvoa += t1a.T + l1a.T
+
+        xt2b += einsum('ma,me->ae', t1b, l1b)
+        dvob -= einsum('mi,ma->ai', xt1b, t1b)
+        dvob -= einsum('ie,ae->ai', t1b, xt2b)
+        dvob += t1b.T + l1b.T
+
+    nmoa = nocca + nvira
+    dm1a = np.zeros((nmoa,nmoa))
+    dm1a[:nocca,:nocca] = dooa + dooa.conj().T
+    dm1a[nocca:,nocca:] = dvva + dvva.conj().T
+    if with_t1:
+        dm1a[:nocca,nocca:] = dvoa.conj().T
+        dm1a[nocca:,:nocca] = (dm1a[:nocca,nocca:].conj().T)
+    dm1a = (dm1a + dm1a.T)/4
+
+    nmob = noccb + nvirb
+    dm1b = np.zeros((nmob, nmob))
+    dm1b[:noccb,:noccb] = doob + doob.conj().T
+    dm1b[noccb:,noccb:] = dvvb + dvvb.conj().T
+    if with_t1:
+        dm1b[:noccb,noccb:] =  dvob.conj().T
+        dm1b[noccb:,:noccb] = (dm1b[:noccb,noccb:].conj().T)
+    dm1b = (dm1b + dm1b.T)/4
+
+    if with_mf:
+        dm1a[np.diag_indices(nocca)] += 1
+        dm1b[np.diag_indices(noccb)] += 1
+
+    if ao_basis:
+        dm1a = dot(emb.mo_coeff[0], dm1a, emb.mo_coeff[0].T)
+        dm1b = dot(emb.mo_coeff[1], dm1b, emb.mo_coeff[1].T)
+
+    # --- Some information:
+    emb.log.debug("Cluster-pairs: total= %d  kept= %d (%.1f%%)", total_xy, kept_xy, 100*kept_xy/total_xy)
+    emb.log.debug("Singular values (α) total= %d  kept= %d (%.1f%%)", total_sv_a, kept_sv_a, 100*kept_sv_a/total_sv_a)
```

Comment:
Only ASCII output please

Reply:
I think UTF-8 has been standard on linux for a while now. Must we support users with computers from the 80s?

Reply:
This has absolutely nothing to do with the OS not supporting UTF-8, but keyboards not having UTF-8 keys